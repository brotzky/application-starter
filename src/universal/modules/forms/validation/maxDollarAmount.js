import numeral from 'numeral';

export default function(max) {
  if (!max) {
    return '';
  }
  return function(value) {
    return value > max
      ? {
          defaultMessage: 'cannot be more than {amount}',
          id: 'form.errors.maxDollarAmount',
          values: { amount: numeral(max).format('$0,00') },
        }
      : undefined;
  }
}
