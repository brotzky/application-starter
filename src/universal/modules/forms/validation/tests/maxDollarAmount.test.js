import maxDollarAmount from '../maxDollarAmount';

const maxOneHundred = maxDollarAmount(100);

describe('[Form Validation] maxDollarAmount', () => {
  describe('maxDollarAmount()', () => {
    it('should return a function', () => {
      expect(maxDollarAmount(100)).toMatchSnapshot();
    });

    it('should return no errors if the number passed is less than the originally specified amount', () => {
      expect(maxOneHundred(0)).toMatchSnapshot();
    });

    it('should return an error object when an amount over the specified is passed in', () => {
      expect(maxOneHundred(200)).toMatchSnapshot();
    });
  });
});
