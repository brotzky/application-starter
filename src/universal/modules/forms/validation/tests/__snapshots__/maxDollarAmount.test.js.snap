// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] maxDollarAmount maxDollarAmount() should return a function 1`] = `[Function]`;

exports[`[Form Validation] maxDollarAmount maxDollarAmount() should return an error object when an amount over the specified is passed in 1`] = `
Object {
  "defaultMessage": "cannot be more than {amount}",
  "id": "form.errors.maxDollarAmount",
  "values": Object {
    "amount": "$100",
  },
}
`;

exports[`[Form Validation] maxDollarAmount maxDollarAmount() should return no errors if the number passed is less than the originally specified amount 1`] = `undefined`;
